### PR TITLE
Propagates UserTable privacy to Table 6021

### LINKED ISSUE #6021: When importing through IMPORT API viz privacy is not set as dataset privacy is set on upload
According to the referenced commit:

https://github.com/CartoDB/cartodb/pull/5022

"the privacy of the viz is the same as whatever the privacy param was set to."

I can't see that behavior on the latest version. Setting a dataset's privacy does work but when create_vis is set to "true" on the import api the resulting visualization is always set as private regardless of the privacy setting specified through the IMPORT API call.

### PR DESCRIPTION
**Pull request by gfiorav**
Fixes #6021

There was a typo where the privacy was changed for `UserTable` but not saved.

To Do:
- [x] CR ( @ethervoid )
- [x] Modify test to check for this
- [x] Staging tests

cc/ @cesarandreslopez

### CODE REVIEW

**Comment by gfiorav** [comment]
Ready for CR @ethervoid :godmode:

File: app/models/table.rb [hunk 1]
```diff
@@ -424,6 +423,8 @@ def after_create
         @user_table.privacy = @data_import.privacy
       end
 
+      save
```
Comment (ethervoid) [inline app/models/table.rb]: Here we are saving but this is the after_create, not the after_save (even there you can't do this). Maybe you want to do `@user_table.save` ?

**Comment by ethervoid** [comment]
Check the comment before deploy

File: app/models/table.rb (hunk 1 above)
Comment (gfiorav) [inline app/models/table.rb]: You're right!

**Pull request merged**